src/panel-chat.tsx

```diff
@@ -112,16 +112,77 @@ class ChatTextEntry extends preact.Component<{room: PSRoom, onMessage: (msg: str
 		elem.focus();
 	};
 	keyPress = (e: KeyboardEvent) => {
+		const cmdKey = (((e.cmdKey || e.metaKey) ? 1 : 0) + (e.ctrlKey ? 1 : 0) === 1) && !e.altKey && !e.shiftKey;
 		let elem = e.currentTarget as HTMLTextAreaElement;
-		if (e.keyCode === 13 && !e.shiftKey) {
+		if (e.key === 'Enter' && !e.shiftKey) {
```

Comment:
`e.key` has pretty bad compatibility; I'd rather not.

Reply:
What does that mean? `e.key` is supported in IE9, and I don't think we support IE8 for non-replay related parts of client (which this isn't).

Reply:
See the notes "Partial support refers to these versions of Firefox returning `'MozPrintableKey'` for all character keys". And "Has non-standard key identifiers and incorrect behaviour with AltGraph."

This table doesn't even cover _all_ the problems with `event.key` – it says Firefox 29+ is standards compliant, and doesn't even mention that it took until Firefox 37+ to return `" "` instead of `"Spacebar"` for the spacebar (IE9-IE11, supported browsers, also have this problem). Other keys with incompatibilities in supported browsers include the arrow keys (`"Left"` vs `"ArrowLeft"`, all the way up to Edge 16), and Esc (`"Escape"` vs `"Esc"`).

None of these incompatibilities exist in `event.keyCode`.

And even if I don't care about compatibility with IE11 (I do), there's still the question of, I don't want to have to learn multiple key APIs just to detect a keypress. The one I know is consistent and works in all browsers; I don't care if it's deprecated; I write other software that does need to support older browsers, and I don't want myself or other contributors to have to learn multiple APIs just because some people want to use the new hot API.

Like, sure, if the new hot API was appreciably better or something, but it clearly isn't.

Reply:
This is actually one of the things that I really am annoyed at MDN for; their advice is "`keyCode` is deprecated; you should use an alternative. No true alternatives exist."

They don't even bother to explain the tradeoffs alternatives have. It's just a vague "you might have compatibility issues sometimes, but we won't tell you which times."

Reply:
Googling for advice on how to handle key presses give [Medium articles](https://medium.com/@uistephen/keyboardevent-key-for-cross-browser-key-press-check-61dbad0a067a) by "experts" who don't know the difference between `preventDefault` and `stopPropagation`:

> I’m curious if anyone knows for sure what the behavior is and how necessary it is to include this. Since it doesn’t hurt anything, I’m including it anyway.

With such glorious recommended code as:

```
var key = event.key || event.keyCode;
if (key === 'Escape' || key === 'Esc' || key === 27) { }
```

Oh yes, this is _so_ much better than `if (event.keyCode === 27) {}`. I wonder why webdevs aren't switching in _droves_ to this _new better API_...

Reply:
```suggestion
		if (e.keyCode === 13 && !e.shiftKey) {
```